rs/canister_sandbox/src/replica_controller/sandboxed_execution_controller.rs

```diff
@@ -1193,6 +1196,21 @@ impl SandboxedExecutionController {
                 metrics
                     .sandboxed_execution_subprocess_memfd_rss_total
                     .set(total_memfd_rss.try_into().unwrap());
+
+                println!("anon rss: {}", total_anon_rss);
+
+                // This is memory-usage based eviction. When the usage is over
+                // the limit, we evict half of the sandboxes between the min and max.
+                if total_anon_rss >= SANDBOX_PROCESSES_MAX_ANON_KIB {
+                    let current_no_sandboxes = max_sandbox_count.min(sandbox_processes.len());
+                    let mut guard = backends.lock().unwrap();
+                    evict_sandbox_processes(
+                        &mut guard,
+                        min_sandbox_count,
+                        (current_no_sandboxes - min_sandbox_count) / 2,
```

Comment:
I'm not sure I understand this. If currently there are 500 sandboxes and min count is 500, this will be `(500 - 500) / 2`, right?

Reply:
this is just for testing purposes, not supposed to be production ready; Also, that condition will be triggered if we have 500 sandboxes that in total use `SANDBOX_PROCESSES_MAX_ANON_KIB`. Not saying that's impossible, but we're trying to iterate fast here and find a proper number of sandboxes to be removed.